digitalocean/datasource_digitalocean_firewall_test.go

```diff
@@ -0,0 +1,86 @@
+package digitalocean
+
+import (
+	"fmt"
+	"github.com/digitalocean/godo"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceDigitalOceanFirewall_Basic(t *testing.T) {
+	fwDataConfig := `
+data "digitalocean_firewall" "foobar" {
+	firewall_id = digitalocean_firewall.foobar.id
+	name = digitalocean_firewall.foobar.name
```

Comment:
We shouldn't require users to enter both the name and id. All we really need is the id as we can get the name with it.